### ISSUE TITLE
[BUG] - [DnD5e] - Creating an Item Pile from a Weapon Raises Multiple Errors and Yields Scene Corruption

### ISSUE THREAD

**Issue by KelSolaar** [fantasycalendar/FoundryVTT-ItemPiles#234]
**Describe the bug**
Hi,

Since the DnD5e 2.1.2? update, creating an Item Pile from a Weapon raises multiple errors and causes scene corruption once reopening. The same occurs with Item Piles 2.4.0.

**To Reproduce**

Create an Item Pile from a SRD Weapon

**Expected behavior**
The Item Pile should be created.

**Screenshots**

Errors upon creating the Item Pile:

```javascript
Foundry VTT | Error thrown in hooked function '' for hook 'createToken'
#call @ foundry.js:727
callAll @ foundry.js:681
callback @ foundry.js:12613
(anonymous) @ foundry.js:12594
_handleCreateEmbeddedDocuments @ foundry.js:12594
_createEmbeddedDocuments @ foundry.js:12499
await in _createEmbeddedDocuments (async)
create @ commons.js:6737
await in create (async)
createDocuments @ commons.js:6110
createEmbeddedDocuments @ commons.js:6328
_createItemPile @ private-api.js:959
await in _createItemPile (async)
_dropItems @ private-api.js:814
dropItems @ socket.js:98
_executeLocal @ socketlib.js:192
executeAsGM @ socketlib.js:90
executeAsGM @ socket.js:171
_dropItem @ private-api.js:1481
await in _dropItem (async)
_dropData @ private-api.js:1357
await in _dropData (async)
(anonymous) @ helpers.js:42
#call @ foundry.js:724
call @ foundry.js:706
_onDrop @ foundry.js:24683
callback @ foundry.js:56837
_handleDrop @ foundry.js:56888
foundry.js:747 Error: Actor5e [6onLVVqcGjnjAXYy] Model Validation Errors
[Actor5e.items]: Model Validation Errors
Model Validation Errors
[BaseItem.system]: Model Validation Errors
[WeaponData.proficient]: may not be undefined
[Detected 2 packages: system:dnd5e, item-piles]
    at SchemaField._validateType (commons.js:4098:15)
    at SchemaField.validate (commons.js:3824:37)
    at Actor5e.validate (commons.js:5492:35)
    at new DataModel (commons.js:5269:26)
    at new Document (commons.js:5811:3)
    at new BaseActor (commons.js:7158:3)
    at new ClientDocumentMixin (foundry.js:13053:5)
    at new Actor (foundry.js:16292:1)
    at new Actor5e (actor.mjs:10:16)
    at TokenDocument5e.getActor (foundry.js:21987:19)
    at get actor [as actor] (foundry.js:21861:44)
    at getActor (utilities.js:9:10)
    at isValidItemPile (pile-utilities.js:24:23)
    at PrivateAPI._onCreateToken (private-api.js:150:10)
    at Object.fn (helpers.js:42:7)
    at #call (foundry.js:724:20)
    at Hooks.callAll (foundry.js:681:17)
    at ClientDatabaseBackend.callback (foundry.js:12613:13)
    at foundry.js:12594:43
    at Array.map (<anonymous>)
    at ClientDatabaseBackend._handleCreateEmbeddedDocuments (foundry.js:12594:33)
    at ClientDatabaseBackend._createEmbeddedDocuments (foundry.js:12499:17)
    at async TokenDocument5e.createDocuments (commons.js:6110:23)
    at async PrivateAPI._createItemPile (private-api.js:959:31)
    at async PrivateAPI._dropItems (private-api.js:814:22)
onError @ foundry.js:747
🎁call_wrapped @ libWrapper-wrapper.js:507
🎁Hooks.onError#lib-wrapper @ listeners.js:137
🎁Hooks.onError#0 @ libWrapper-wrapper.js:187
#call @ foundry.js:728
callAll @ foundry.js:681
callback @ foundry.js:12613
(anonymous) @ foundry.js:12594
_handleCreateEmbeddedDocuments @ foundry.js:12594
_createEmbeddedDocuments @ foundry.js:12499
await in _createEmbeddedDocuments (async)
create @ commons.js:6737
await in create (async)
createDocuments @ commons.js:6110
createEmbeddedDocuments @ commons.js:6328
_createItemPile @ private-api.js:959
await in _createItemPile (async)
_dropItems @ private-api.js:814
dropItems @ socket.js:98
_executeLocal @ socketlib.js:192
executeAsGM @ socketlib.js:90
executeAsGM @ socket.js:171
_dropItem @ private-api.js:1481
await in _dropItem (async)
_dropData @ private-api.js:1357
await in _dropData (async)
(anonymous) @ helpers.js:42
#call @ foundry.js:724
call @ foundry.js:706
_onDrop @ foundry.js:24683
callback @ foundry.js:56837
_handleDrop @ foundry.js:56888
commons.js:6946 Foundry VTT | Created Token with id [dc6PtMAqiYwBVqHX] in parent Scene [7tLl4tAU8jqdXVRH]
commons.js:4098 Uncaught (in promise) Error: Actor5e [6onLVVqcGjnjAXYy] Model Validation Errors
[Actor5e.items]: Model Validation Errors
Model Validation Errors
[BaseItem.system]: Model Validation Errors
[WeaponData.proficient]: may not be undefined
[Detected 1 package: system:dnd5e]
    at SchemaField._validateType (commons.js:4098:15)
    at SchemaField.validate (commons.js:3824:37)
    at Actor5e.validate (commons.js:5492:35)
    at new DataModel (commons.js:5269:26)
    at new Document (commons.js:5811:3)
    at new BaseActor (commons.js:7158:3)
    at new ClientDocumentMixin (foundry.js:13053:5)
    at new Actor (foundry.js:16292:1)
    at new Actor5e (actor.mjs:10:16)
    at TokenDocument5e.getActor (foundry.js:21987:19)
    at get actor [as actor] (foundry.js:21861:44)
    at get actor [as actor] (foundry.js:46016:26)
    at Token5e.drawBars (foundry.js:46749:16)
    at Token5e._draw (foundry.js:46555:10)
    at async Token5e.draw (foundry.js:25199:5)
```

Corrupted scene upon reloading:

![image](https://user-images.githubusercontent.com/99779/212279908-78be2350-ffb2-48f8-b926-91a9eb774c05.png)

**Setup:**

- Foundry Version: 10.291
- System Version: DnD5e 2.1.1
- Item Piles Version: 2.3.11, 2.4.0
- Browser & version: N/A

**Active modules:**

- Item Piles
- libwrapper
- socketlib

**Additional context**
N/A

**Comment by action-jay** [comment]
I'm troubleshooting this at the moment too. 2.1 did a big change to data validation and warned about old actors or items causing problems. It promised to mark any worlds triggering errors as 'legacy' and making the validation more permissive/backwards compatible, but item piles' data seems to have snuck through.

In my testing, my reliable trigger item is the playing card set, which would also be affected by a character's proficiencies mentioned in the error messages.

- Deleting the Default Item Pile and/or Chest actor lets me get back into an affected scene - now I have unlinked tokens that at least still have the item data until this if figured out

- Setting the affected token to represent a different actor breaks the scene again, even with modules disabled. Dropping a new, empty item pile onto the scene and then adding the item after the fact works fine (yay!). This tells me there's is something to do with the data specifically when it got written in by dropping the item on the canvas.

- I also found that after deleting the original 'default item pile', I can't drop items onto the canvas -at all- now and the creation of a new Default Item Pile actor fails even when using what should be a relatively safe item, like a piece of loot with no other properties. That stymies my testing for now. But! I hope these findings help.
